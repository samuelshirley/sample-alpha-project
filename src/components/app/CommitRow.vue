<template>
  <tr
    :key="commit.sha"
    @click="goToDetails"
  >
    <td v-text="commit.author.login" />
    <td v-text="commit.sha" />
    <td v-text="commit.commit.message" />
    <td v-text="commit.commit.author.date" />
    <!-- <td v-text="'Author'" />
    <td v-text="'abcdefabcdefabcdefabcdefabcdefabcdef'" />
    <td v-text="'Commit Message'" />
    <td v-text="'2019-11-01T00:00:00Z'" /> -->
  </tr>
</template>

<script>
export default {
  props: {
    commit: {
      type: Object,
      default: () => ({}),
    },
  },
  methods: {
    goToDetails () {
      const query = {}
      query.page = this.$route.query.page || 1
      query.per_page = this.$route.query.per_page || 10
      this.$router.push({ path: `/commits/${this.commit.sha}`, query })
    },
  },
}
</script>

<style>

</style>
