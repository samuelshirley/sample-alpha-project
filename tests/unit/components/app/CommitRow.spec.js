import { shallowMount } from '@vue/test-utils'
import CommitRow from '@/components/app/CommitRow.vue'

const commit = {
  sha: '175b9acfd438be86de7687bc48bf9d84c0ee4949',
  author: {
    login: 'TestUsername',
  },
  commit: {
    message: 'Test Commit Message',
    author: {
      date: '2019-03-25T15:49:30.674Z',
    },
  },
}

describe('CommitRow.vue', () => {
  it('renders correctly when commit passed', () => {
    const wrapper = shallowMount(CommitRow, {
      propsData: {
        commit,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders correctly when commit passed', () => {
    const pushSpy = jest.fn()
    const wrapper = shallowMount(CommitRow, {
      propsData: {
        commit,
      },
      mocks: {
        $router: {
          push: pushSpy,
        },
        $route: {
          query: {},
        },
      },
    })
    wrapper.vm.goToDetails()
    expect(pushSpy).toHaveBeenCalledWith({ path: `/commits/${commit.sha}`, query: { page: 1, per_page: 10 } })
  })
})
