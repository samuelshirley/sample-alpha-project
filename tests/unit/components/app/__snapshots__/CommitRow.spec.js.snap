// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitRow.vue renders correctly when commit passed 1`] = `
<tr>
  <td>TestUsername</td>
  <td>175b9acfd438be86de7687bc48bf9d84c0ee4949</td>
  <td>Test Commit Message</td>
  <td>2019-03-25T15:49:30.674Z</td>
</tr>
`;
